### PR TITLE
Unbind ATL lib on Windows

### PR DESCRIPTION
**Pull request by sarakusha**
[windows-build-tools](https://github.com/felixrieseberg/windows-build-tools) doesn't install ATL, so you can't compile this package until you install ATL manually.

### CODE REVIEW

File: src/detection_win.cpp [hunk 1]
```diff
@@ -379,28 +382,28 @@ void BuildInitialDeviceList() {
...
 
 	SP_DEVINFO_DATA* pspDevInfoData = (SP_DEVINFO_DATA*) HeapAlloc(GetProcessHeap(), 0, sizeof(SP_DEVINFO_DATA));
-	pspDevInfoData->cbSize = sizeof(SP_DEVINFO_DATA);
-	for(int i=0; DllSetupDiEnumDeviceInfo(hDevInfo, i, pspDevInfoData); i++) {
-		DWORD nSize=0 ;
-		TCHAR buf[MAX_PATH];
+	if (pspDevInfoData) {
+        pspDevInfoData->cbSize = sizeof(SP_DEVINFO_DATA);
+        for(int i=0; DllSetupDiEnumDeviceInfo(hDevInfo, i, pspDevInfoData); i++) {
+            DWORD nSize=0 ;
```
Comment (MadLittleMods) [inline src/detection_win.cpp]: Let's opt for keeping the tabs around (mixing in spaces currently)

It's also making the diff hard to read

Comment (sarakusha) [reply src/detection_win.cpp]: It's my IDE. I just wanted to indent block

**Comment by sarakusha** [comment]
I have already fixed the formatting of the code, how about commit PR.

File: src/detection_win.cpp [hunk 2]
```diff
@@ -379,28 +382,28 @@ void BuildInitialDeviceList() {
 	}
 
 	SP_DEVINFO_DATA* pspDevInfoData = (SP_DEVINFO_DATA*) HeapAlloc(GetProcessHeap(), 0, sizeof(SP_DEVINFO_DATA));
-	pspDevInfoData->cbSize = sizeof(SP_DEVINFO_DATA);
-	for(int i=0; DllSetupDiEnumDeviceInfo(hDevInfo, i, pspDevInfoData); i++) {
-		DWORD nSize=0 ;
-		TCHAR buf[MAX_PATH];
+	if (pspDevInfoData) {
```
Comment (MadLittleMods) [inline src/detection_win.cpp]: Why can `pspDevInfoData` be undefined now?

File: src/detection_win.cpp [hunk 3]
```diff
@@ -468,79 +471,77 @@ void UpdateDevice(PDEV_BROADCAST_DEVICEINTERFACE pDevInf, WPARAM wParam, DeviceS
...
-				isAdded = false;
-			}
 
-			break;
+				break;
+			}
 		}
-	}
 
-	if (pspDevInfoData) {
```
Comment (MadLittleMods) [inline src/detection_win.cpp]: Why is this check removed but we are adding if statements around other areas?

**Comment by MadLittleMods** [comment]
Seems to build fine for me and ran the tests(`npm test`)

 - Windows 10
 - `node@10.15.1` (`node -v`)
 - `node-gyp`: v3.8.0 (`node-gyp -v`)
 - Python 2.7
 - `C:\Program Files (x86)\Microsoft Visual Studio\2017\Community` (`node-gyp configure --verbose`)
    - `Microsoft.VisualStudio.Component.Windows10SDK.17763`
    - `Microsoft.VisualStudio.Component.VC.Tools.x86.x64
    - `Microsoft.VisualStudio.VC.MSBuild.Base`
    - `Using this installation with Windows 10 SDK`

---

@sarakusha Have you ran through the tests on your machine? Can you share your environment details for reference?

File: src/detection_win.cpp (hunk 3 above)
Comment (sarakusha) [reply src/detection_win.cpp]: Previously, they first used a pointer, then checked for null before release. Now, first, check for null, then use and release

Comment (sarakusha) [reply src/detection_win.cpp]: I use CLion IDE, which has formatted the source code, so there are a lot of differences.

File: src/detection_win.cpp (hunk 2 above)
Comment (sarakusha) [reply src/detection_win.cpp]: > If the function fails and you have not specified HEAP_GENERATE_EXCEPTIONS, the return value is NULL.
And below we check the pointer before release. I moved the check here.

**Comment by pauloavelar** [comment]
Sorry to jump in, but is there a timeline for this feature to be merged and released?
I am currently benchmarking libraries for listing USB devices in Electron but don't want to add a full VS installation requirement. I can't build on Windows right now because my env lacks "atlstr.h", which I suppose this PR fixes.

**Comment by webees** [comment]
@pauloavelar use `vs_buildtools__1891780700.1567674472.exe`  to check ATL.

**Comment by MadLittleMods** [comment]
Sorry for the delay 🥶

This fix is released in [`usb-detection@4.5.0`](https://github.com/MadLittleMods/node-usb-detection/blob/master/CHANGELOG.md#450---2019-10-6) :tada:

Thanks for the fix @sarakusha ❤️

**Pull request merged**